src/core/observable.ts

```diff
@@ -224,8 +232,48 @@ export function propagateMaybeChanged(observable: IObservable) {
         const d = observers[i]
         if (d.dependenciesState === IDerivationState.UP_TO_DATE) {
             d.dependenciesState = IDerivationState.POSSIBLY_STALE
+            if (d.isTracing !== TraceMode.NONE) {
+                logTraceInfo(d, observable)
+            }
             d.onBecomeStale()
         }
     }
     // invariantLOS(observable, "maybe end");
 }
+
+function logTraceInfo(derivation: IDerivation, observable: IObservable) {
+    console.log(
+        `[mobx.trace] '${derivation.name}' is invalidated due to a change in: '${observable.name}'`
+    )
+    if (derivation.isTracing === TraceMode.BREAK) {
+        const lines = []
+        printDepTree(getDependencyTree(derivation), lines, 1)
+
+        // prettier-ignore
+        new Function(
+`debugger;
+/*
+Tracing '${derivation.name}'
```

Comment:
I've not seen this pattern before; did you copy it from somewhere else (and does it have a name), or is this your own invention?